commands/interpolate_test.go

```diff
@@ -0,0 +1,222 @@
+package commands_test
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"runtime"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gbytes"
+	"github.com/onsi/gomega/gexec"
+	. "github.com/onsi/gomega/ghttp"
+)
+
+var (
+	session      *gexec.Session
+	templateFile *os.File
+	templateText string
+	err          error
+)
+
+var _ = Describe("interpolate", func() {
+	BeforeEach(func() {
+		login()
+		templateFile, err = ioutil.TempFile("", "credhub_test_interpolate_template_")
+	})
+
+	Context("shared command examples", func() {
+		templateFile, err = ioutil.TempFile("", "credhub_test_interpolate_template_")
+		templateFile.WriteString("---")
+		ItAutomaticallyLogsIn("GET", "get_response.json", "/api/v1/data", "interpolate", "-f", templateFile.Name())
+
+		ItBehavesLikeHelp("interpolate", "interpolate", func(session *gexec.Session) {
+			Expect(session.Err).To(Say("Usage"))
+			if runtime.GOOS == "windows" {
+				Expect(session.Err).To(Say("credhub-cli.exe \\[OPTIONS\\] interpolate \\[interpolate-OPTIONS\\]"))
+			} else {
+				Expect(session.Err).To(Say("credhub-cli \\[OPTIONS\\] interpolate \\[interpolate-OPTIONS\\]"))
+			}
+		})
+		ItRequiresAuthentication("interpolate", "-f", "testinterpolationtemplate.yml")
+		ItRequiresAnAPIToBeSet("interpolate", "-f", "testinterpolationtemplate.yml")
+	})
+
+	It("queries for string creds and prints them in the template as strings", func() {
```

Comment:
Could we group the following three It's under a context as they are somewhat related.

Reply:
Sure! How about `Describe("Interpolating various types of credentials"` sound?